### ISSUE TITLE
logsumexp and negative exp's

### ISSUE THREAD

**Issue by spinkney** [stan-dev/math#2127]
This is a feature request to allow "subtraction" for example

log(exp(a) - exp(b)) 

so that the function logsumexp([a, b], scale = [1, -1]) or something similar. Would be similar to `scipy` implementation of logsumexp 

https://github.com/scipy/scipy/blob/f2ec91c4908f9d67b5445fbfacce7f47518b35d1/scipy/misc/common.py#L18

**Comment by andrjohns** [comment]
Does the [log_diff_exp](https://github.com/stan-dev/math/blob/develop/stan/math/prim/fun/log_diff_exp.hpp) function cover what you're after?

**Comment by spinkney** [comment]
It would if it had a vector signature like `log_sum_exp`. Any chance that can get added?

**Comment by rok-cesnovar** [comment]
Yes, @andrjohns has added vectorized versions of: 
```
 bessel_first_kind
 bessel_second_kind
 beta
 binary_log_loss
 binomial_coefficient_log
 choose
 falling_factorial
 fdim
 fmax
 fmin
 fmod
 gamma_p
 gamma_q
 hypot
 lbeta
 ldexp
 lmgamma
 log_diff_exp
 log_falling_factorial
 log_inv_logit_diff
 log_modified_bessel_first_kind
 log_rising_factorial
 modified_bessel_first_kind
 modified_bessel_second_kind
 multiply_log
 owens_t
 pow
 rising_factorial
```
These are available on develop and will be available in the upcoming 2.25 release (next week will be the release candidate).

**Comment by andrjohns** [comment]
I think they meant ```log_diff_exp(vector)``` rather than ```log_diff_exp(vector, real)``` , is that right @spinkney?

**Comment by andrjohns** [comment]
Yep, this can be defined for vectors/matrices like ```log_sum_exp```, I put together a rough prototype on godbolt as a proof-of-concept: https://godbolt.org/z/xET4cE

I can work this up later next week

**Comment by spinkney** [comment]
Actually what I want is an elementwise log_diff_exp. I'm not sure if that would even be faster than a loop though.

**Comment by andrjohns** [comment]
Can you clarify that a little? What signature are you after? Something like ```log_diff_exp(matrix, real)```?

**Comment by spinkney** [comment]
For example

```
vector[N] theta1;
vector[N] theta2
target += log(inv_logit(theta1) - inv_logit(theta2));
```
After a bit of algebra that becomes
```
numerator = log(exp(-theta1) - exp(-theta2);
denom = log(1 + exp(-theta1) + exp(-theta2) + exp(-theta1 - theta2));
```
so the numerator is log_diff_exp(-theta1, -theta2) and the denom is log_sum_exp(0, -theta1, -theta2, -(theta1 + theta2).

**Comment by andrjohns** [comment]
Yep, the elementwise ```log_diff_exp(vector, vector)``` will be available in 2.25.

Incidentally, where you have:
```
log(inv_logit(theta1) - inv_logit(theta2));
```

In 2.25, there will also be an elementwise ```log_inv_logit_diff```, so you can just write:
```
log_inv_logit_diff(theta1, theta2);
```
